### PR TITLE
Plotly - add missing property uid

### PR DESCRIPTION
**Pull request by mrtnbrst**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
> I only formatted the code, again a file changed, no actual change to the code was made

- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>

https://github.com/plotly/plotly.js/blob/ab5e16afc34fa4f22efffef129f811381580e0fe/src/plots/attributes.js#L109 -> uid
Mapbox has uid: https://github.com/plotly/plotly.js/blob/96c28159c06a3af3fe551175db89599894efbf82/src/plots/mapbox/mapbox.js#L35

https://plotly.com/python-api-reference/generated/plotly.graph_objects.Figure.html
> uid – Assign an id to this trace, Use this to provide object constancy between traces during animations and transitions.

Traces have uid:
https://github.com/plotly/plotly.js/blob/19030614649428a0341d28402178bdd32fc75357/src/plot_api/plot_api.js#L679

https://github.com/plotly/plotly.js/blob/ab5e16afc34fa4f22efffef129f811381580e0fe/src/plots/attributes.js#L224
Description of uirevision (already in type definition) using the uid:
>'Trace changes are tracked by `uid`, which only falls back on trace',
            'index if no `uid` is provided. So if your app can add/remove traces',
            'before the end of the `data` array, such that the same trace has a',
            'different index, you can still preserve user-driven changes if you',
            'give each trace a `uid` that stays with it as it moves.'

I added to uid to the places where the uirevision was already there / where I found hints about the uid being available.

**I only manually tested the uid changes for my personal use case, a scatter plot (relevant type ScatterData = PlotData), I am not entirely sure about all the changes I did.**

### CODE REVIEW

**Comment by mrtnbrst** [comment]
I guess I shouldn't be an owner, I misunderstood the meaning of it.

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## plotly.js ([<kbd>unpkg</kbd>](https://unpkg.com/browse/plotly.js@latest/))
was missing the following properties:
1. version
1. Fx
1. Snapshot
1. PlotSchema
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 20003c9c94883b2dac89c990ec61a014e4c16892
</p>

File: types/plotly.js/test/core-tests.ts
Comment (brammitch) [inline types/plotly.js/test/core-tests.ts:1]: Formatting-only changes look good to me. Matches what I see on my local copy.

**Review by brammitch — APPROVED** [review]
UID changes look good to me.

**Review by brammitch — APPROVED** [review]

**Comment by mrtnbrst** [comment]
Ready to merge

**Pull request merged**